src/consensus/encode.rs

```diff
@@ -517,4 +533,55 @@ mod tests {
         assert_eq!(vec![0b1010_1100, 0b0000_0010], serialize(&VarInt(300)));
         assert_eq!("80e497d012", hex::encode(serialize(&VarInt(5000000000))));
     }
+
+    #[test]
+    fn deserialize_vec() {
+        // First byte is len = 8
+        let vec = [0x08u8, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08];
+        let data = deserialize::<Vec<u8>>(&vec).unwrap();
+        assert_eq!(data, vec[1..]);
+
+        let tx_in = serialize(&TxIn::Gen { height: VarInt(1) });
+        let mut vec = vec![0x08u8];
+        for _ in 0..8 {
+            vec.extend(tx_in.clone());
+        }
+        let tx_ins = deserialize::<Vec<TxIn>>(&vec).unwrap();
+        assert!(tx_ins
+            .iter()
+            .all(|t| matches!(t, TxIn::Gen { height: VarInt(1) })));
```

Comment:
@h4sh3d any reason why `TxIn` can't implement `PartialEq`? That would allow us to assert the same way we do on line 542.

Reply:
There is no particular reasons for not having `PartialEq` implemented on `TxIn`, `KeyImage` can have it and `hash::Hash` too.

Reply:
Added to `TxIn` and `KeyImage` (Already implemented for `Hash`)